edx_solutions_organizations/views.py

```diff
@@ -318,25 +318,16 @@ def courses(self, request, pk):  # pylint: disable=W0613
         """
         exclude_admins = str2bool(request.query_params.get('exclude_admins'))
         organization = self.get_object()
-        organization_courses = []
-        organization_course_ids = []
         roles_to_exclude = []
         if exclude_admins:
-            organization_courses = CourseEnrollment.objects\
-                .filter(user__organizations=organization, is_active=True)\
-                .order_by('course_id').distinct().values_list('course_id', flat=True)
-            for course_id in organization_courses:
-                organization_course_ids.append(course_id.to_deprecated_string())
-            organization_courses = map(get_course_key, filter(None, organization_course_ids))
             roles_to_exclude = [CourseInstructorRole.ROLE, CourseStaffRole.ROLE, CourseObserverRole.ROLE, CourseAssistantRole.ROLE]
 
-        enrollment_qs = CourseEnrollment.objects\
-            .filter(user__organizations=organization, is_active=True)\
-            .exclude(
-                user_id__in=CourseAccessRole.objects.filter(
-                    course_id__in=organization_courses, role__in=roles_to_exclude
-                ).values_list('user_id', flat=True)
-            ).values_list('course_id', 'user_id')
+        enrollment_qs = CourseEnrollment.objects.filter(user__organizations=organization, is_active=True).annotate(
+            user_with_role=Exists(
+                CourseAccessRole.objects.filter(
+                    course_id=OuterRef('course_id'), role__in=roles_to_exclude, user_id=OuterRef('user_id'))
```

Comment:
We considered using `OuterRef ` in past. But we are on `Django==1.8.19` and it doesn't have `OuterRef`.